src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> header;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            header = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, header);
+            header.prev = newNode;
+            header = newNode;
+        } else {
+            Node<T> currentPosition = findNode(index);
+            Node<T> newNode = new Node<>(currentPosition.prev, value, currentPosition);
+            currentPosition.prev.next = newNode;
+            currentPosition.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null) {
+            throw new NullPointerException("List is null");
+        }
+        for (T items : list) {
+            add(items);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
```

Comment:
you are calling this already in `findNode()`

Reply:
done